src/CommunityToolkit.Aspire.Hosting.Java/JavaAppHostingExtension.Executable.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                return;
+            }
+
+            var logger = e.Services.GetRequiredService<ResourceLoggerService>().GetLogger(javaAppResource);
+            var notificationService = e.Services.GetRequiredService<ResourceNotificationService>();
+
+            await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+            {
+                State = new("Building Maven project", KnownResourceStates.Starting)
+            }).ConfigureAwait(false);
+
+            logger.LogInformation("Building Maven project");
+
+            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+            var mvnw = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = isWindows ? "cmd" : "sh",
+                    Arguments = isWindows ? $"/c {mavenOptions.Command} {string.Join(" ", mavenOptions.Args)}" : $"./{mavenOptions.Command} {string.Join(" ", mavenOptions.Args)}",
+                    WorkingDirectory = mavenOptions.WorkingDirectory ?? javaAppResource.WorkingDirectory,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true,
+                    UseShellExecute = false,
+                }
+            };
+
+            mvnw.OutputDataReceived += async (sender, args) =>
+            {
+                if (!string.IsNullOrWhiteSpace(args.Data))
+                {
+                    await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+                    {
+                        State = new(args.Data, KnownResourceStates.Starting)
+                    }).ConfigureAwait(false);
+
+                    logger.LogInformation("{Data}", args.Data);
+                }
+            };
+
+            mvnw.ErrorDataReceived += async (sender, args) =>
+            {
+                if (!string.IsNullOrWhiteSpace(args.Data))
+                {
+                    await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+                    {
+                        State = new(args.Data, KnownResourceStates.FailedToStart)
+                    }).ConfigureAwait(false);
+
+                    logger.LogError("{Data}", args.Data);
+                }
+            };
+
+            mvnw.Start();
+            mvnw.BeginOutputReadLine();
+            mvnw.BeginErrorReadLine();
+
+            await mvnw.WaitForExitAsync(ct).ConfigureAwait(false);
+
+            if (mvnw.ExitCode != 0)
+            {
+                await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+                {
+                    State = new($"mvnw exited with {mvnw.ExitCode}", KnownResourceStates.FailedToStart)
+                }).ConfigureAwait(false);
+
+                throw new InvalidOperationException($"mvnw build failed with exit code {mvnw.ExitCode}");
```

Comment:
Why throw?

Reply:
It's how I've seen other hooks implemented in the past.

Reply:
Where does this show up? Wouldn't it be better to show it in the dashboard? Also what happens when the build fails? Do I need to restart the entire apphost to rebuild or can I trigger it from the dashboard? (custom command)

Worth thinking through that I think.

Reply:
The notification service reporting shows up on the dashboard and I think the exception bubbles to the console of the resource that it was relative for.

A custom command would be a good idea to run a rebuild, I'll add that.